presto-hive/src/main/java/com/facebook/presto/hive/HiveSplit.java

```diff
@@ -226,6 +239,12 @@ public boolean isS3SelectPushdownEnabled()
         return extraFileInfo;
     }
 
+    @Override
+    public int getAffinityIdentifier()
```

Comment:
Yeah two hash function sounds reasonable.

`List` seems better than `PriorityQueue`. If we want to indicate that `candidates` is sorted, we should probably rename the variable to `sortedCandidates` or `orderedCandidates`.

A side question is that this interface kind of assumes that we would always pass in the all nodes. Would this conflict with the resource manager and multiple coordinator initiative? Should we consider making the interface align with it given the fact that we might not be able to get a full list of candidate nodes here? If that's the case, we probably need to know the total number of nodes, and the list of candidates (maybe they should be in a map?) should have associated integer ids. Maybe this is a bit overthinking and premature optimization, but want to think out loud in case this was missed.

Reply:
That is a reasonable assumption. In the worst case, we might need to pass in an ordered map with all nodes as the keys as active/inactive as values.